ta/pkcs11/src/persistent_token.c

```diff
@@ -73,6 +73,64 @@ static TEE_Result open_pin_file(struct ck_token *token,
 					0, out_hdl);
 }
 
+static enum pkcs11_rc do_hash(uint32_t user, const uint8_t *pin,
+			      size_t pin_size, uint32_t salt,
+			      uint8_t hash[TEE_MAX_HASH_SIZE])
+{
+	TEE_Result res = TEE_SUCCESS;
+	TEE_OperationHandle oh = TEE_HANDLE_NULL;
+	size_t sz = TEE_MAX_HASH_SIZE;
+
+	res = TEE_AllocateOperation(&oh, TEE_ALG_SHA256, TEE_MODE_DIGEST, 0);
+	if (res)
+		return PKCS11_CKR_GENERAL_ERROR;
```

Comment:
don't you prefer to report at least of out-of-memory status when allocation fails for that.

Reply:
OK, I'll update with `tee2pkcs_error()`.